docs/user-guide/replicasets.md

```diff
@@ -0,0 +1,95 @@
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Replica Set
+
+## What is a _replica set_?
+
+Replica set is the next generation replication controller. The only difference between a _replica set_ and a [_replication controller_](replication-controller.md) right now is the selector support. Replica set supports the new set-based selector requirements as described in the [labels user guide](labels.md#label-selectors) whereas a replication controller only supports equality-based selector requirements.
+
+While replica sets can be used independently, today it's mainly used by [deployments](deployments.md) as a mechanism to orchestrate pod creation, deletion and updates. When you use deployments you don't have to worry about manging the replica sets that they create. Deployments own and manage their replica sets.
+
+### Example
+
+{% include code.html language="yaml" file="replicasets/frontend.yaml" ghlink="/docs/user-guide/replicasets/frontend.yaml" %}
+
+Saving this config into `frontend.yaml` and submitting it to a Kubernetes cluster should create the defined replica set and the pods that it manages.
+
+```sh
+$ kubectl create -f frontend.yaml 
+replicaset "frontend" created
+$ kubectl describe rs/frontend
+Name:		frontend
+Namespace:	default
+Image(s):	gcr.io/google_samples/gb-frontend:v3
+Selector:	tier=frontend,tier in (frontend)
+Labels:		app=guestbook,tier=frontend
+Replicas:	3 current / 3 desired
+Pods Status:	3 Running / 0 Waiting / 0 Succeeded / 0 Failed
+No volumes.
+Events:
+  FirstSeen	LastSeen	Count	From				SubobjectPath	Type		Reason			Message
+  ---------	--------	-----	----				-------------	--------	------			-------
+  1m		1m		1	{replicaset-controller }			Normal		SuccessfulCreate	Created pod: frontend-qhloh
+  1m		1m		1	{replicaset-controller }			Normal		SuccessfulCreate	Created pod: frontend-dnjpy
+  1m		1m		1	{replicaset-controller }			Normal		SuccessfulCreate	Created pod: frontend-9si5l
+$ kubectl get pods
+NAME             READY     STATUS    RESTARTS   AGE
+frontend-9si5l   1/1       Running   0          1m
+frontend-dnjpy   1/1       Running   0          1m
+frontend-qhloh   1/1       Running   0          1m
+```
+
+## Replica set as an Horizontal Pod Autoscaler target
+
+A replica set can also be a target for [Horizontal Pod Autoscalers (HPA)](horizontal-pod-autoscaler.md). Here is an example HPA targeting the replica set we created in the previous example.
```

Comment:
@madhusudancs I might be wrong, but I believe it would provide more clarity to users if the difference is noted on `hpa` with `ReplicaSet` during `rolling-update` (wrt `ReplicationController`). For example, with a `ReplicationController` + `hpa`, the user cannot perform a `rolling-update`. How does `ReplicaSet` differ?